Assets/Scripts/Inventory/Custom/Hands.cs

```diff
@@ -22,41 +20,31 @@ public class Hands : Interaction, Tool
         [System.NonSerialized]
         public int selectedHand = 0;
         public event OnHandChange onHandChange;
-
-        /**
-     * The default hand interaction when no object is present.
-     * Note: This could be moved into a different class if this one gets too cluttered
-     */
-        public void Interact(RaycastHit hit, bool secondary)
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="target"></param>
+        public void Pickup(GameObject target)
         {
-            // All actions just use primary mouse.
-            if(secondary)
-                return;
-
-            if ((hit.point - transform.position).magnitude > handRange)
-                return;
-
-            // To pick up an item, we need to be clicking a free item object with an empty hand
             if (GetItemInHand() == null)
             {
-                var gameObject = hit.collider.gameObject;
-
-                // If an item pick it up
-                Item item = gameObject.GetComponent<Item>();
-                if(item && item.transform.parent == null)
-                    inventory.CmdAddItem(hit.collider.gameObject, handContainer.gameObject, handSlots[selectedHand]);
-
-                // If it has a container, open the container
-                Container container = gameObject.GetComponent<Container>() ?? gameObject.GetComponentInParent<Container>();
-                if(container)
-                    CmdConnectContainer(container.gameObject);
+                inventory.CmdAddItem(target, handContainer.gameObject, handSlots[selectedHand]);
             }
             else
             {
-                // Drop the item currently being held.
-                inventory.CmdPlaceItem(handContainer.gameObject, handSlots[selectedHand], hit.point + new Vector3(0f, 0.2f), new Quaternion());
+                Debug.LogWarning("Trying to pick up with a non-empty hand");
```

Comment:
Is item on item interaction handled or going to be handled somewhere else?

Reply:
Item on item interactions are handled by events, by that when you use an item, you send an event to your held item, which may then forward events to the thing being interacted with